features/networking/ovn.feature

```diff
@@ -0,0 +1,41 @@
+Feature: OVN related networking scenarios
+
+  # @author anusaxen@redhat.com
+  # @case_id OCP-29954
+  @admin
+  @destructive
+  Scenario: Creating a resource in Kube API should be synced to OVN NB db correctly even post NB db crash too
+    Given the env is using "OVNKubernetes" networkType
+    Given I have a project
+    And I obtain test data file "networking/list_for_pods.json"
+    When I run the :create client command with:
+      | f | list_for_pods.json |
+    Then the step should succeed
+    Given 2 pods become ready with labels:
+      | name=test-pods |
+    And evaluation of `pod(0).name` is stored in the :pod1_name clipboard
+    And evaluation of `pod(1).name` is stored in the :pod2_name clipboard
+    #Checking whether Kube API data is synced on OVN NB db which in this case are couple of pods
+    Given I store the ovnkube-master "north" leader pod in the clipboard
+    And evaluation of `pod.ip_url` is stored in the :OVN_NB_leader_IP clipboard
+    And evaluation of `pod.node_name` is stored in the :OVN_NB_leader_node clipboard
+    And admin executes on the pod:
+      | bash | -c | ovn-nbctl -p /ovn-cert/tls.key -c /ovn-cert/tls.crt -C /ovn-ca/ca-bundle.crt --db ssl:<%= cb.OVN_NB_leader_IP %>:9641 list logical_switch_port |
```

Comment:
If we can select a container to run on and use the `nbdb` container, then we wouldn't need to use the SSL options.

e.g.

`oc exec -c nbdb ovnkube-master-4cjwf -- ovn-nbctl list logical_switch_port`

but unfortunately none of the pod execute commands let you specify a container.  I was going to look into adding this somehow.

Reply:
This works in a quick test

```diff
diff --git a/features/networking/ovn.feature b/features/networking/ovn.feature
index d7ea10ef..2a148407 100644
--- a/features/networking/ovn.feature
+++ b/features/networking/ovn.feature
@@ -19,8 +19,8 @@ Feature: OVN related networking scenarios
     Given I store the ovnkube-master "north" leader pod in the clipboard
     And evaluation of `pod.ip_url` is stored in the :OVN_NB_leader_IP clipboard
     And evaluation of `pod.node_name` is stored in the :OVN_NB_leader_node clipboard
-    And admin executes on the pod:
-      | bash | -c | ovn-nbctl -p /ovn-cert/tls.key -c /ovn-cert/tls.crt -C /ovn-ca/ca-bundle.crt --db ssl:<%= cb.OVN_NB_leader_IP %>:9641 list logical_switch_port |
+    And admin executes on the pod "nbdb" container:
+      | bash | -c | ovn-nbctl  list logical_switch_port |
     Then the step should succeed
     And the output should contain:
       | <%= cb.pod1_name %> |
@@ -33,8 +33,8 @@ Feature: OVN related networking scenarios
     #Making sure the pod entries are synced again when NB db is re-created
     Given I store the ovnkube-master "north" leader pod in the clipboard
     And evaluation of `pod.ip_url` is stored in the :New_OVN_NB_leader_IP clipboard
-    Given admin executes on the pod:
-      | bash | -c | ovn-nbctl -p /ovn-cert/tls.key -c /ovn-cert/tls.crt -C /ovn-ca/ca-bundle.crt --db ssl:<%= cb.New_OVN_NB_leader_IP %>:9641 list logical_switch_port |
+    Given admin executes on the pod "nbdb" container:
+      | bash | -c | ovn-nbctl list logical_switch_port |
     Then the step should succeed
     And the output should contain:
       | <%= cb.pod1_name %> |
diff --git a/features/step_definitions/pod.rb b/features/step_definitions/pod.rb
index a98c32b1..9e6a2b75 100644
--- a/features/step_definitions/pod.rb
+++ b/features/step_definitions/pod.rb
@@ -213,7 +213,7 @@ end

 # args can be a table where each cell is a command or an argument, or a
 #   multiline string where each line is a command or an argument
-When /^(I execute|admin executes) on the(?: "(.+?)")? pod:$/ do |by, pod_name, raw_args|
+When /^(I execute|admin executes) on the(?: "(.+?)")? pod(?: "(.+?)" container)?:$/ do |by, pod_name, container, raw_args|
   _user = by.split.first == "admin" ? admin : user
   if raw_args.respond_to? :raw
     # this is table, we don't mind dimentions used by user
@@ -223,7 +223,7 @@ When /^(I execute|admin executes) on the(?: "(.+?)")? pod:$/ do |by, pod_name, r
     args = raw_args.split("\n").map(&:strip)
   end

-  @result = pod(pod_name).exec(*args, as: _user)
+  @result = pod(pod_name).exec(*args, as: _user, container: container)
 end

 # wrapper around  oc logs, keep executing the command until we have an non-empty response
```

Reply:
err, it seems to work on the default `northd` container as well.  I guess as long as we run on the master we don't need to use the SSL.


```bash
$ oc exec  ovnkube-master-4cjwf -- ovn-nbctl list logical_switch_port
Defaulting container name to northd.
Use 'oc describe pod/ovnkube-master-4cjwf -n openshift-ovn-kubernetes' to see all of the containers in this pod.
_uuid               : b5f38583-6461-4d27-93eb-a6027f8c28ee
addresses           : [dynamic]
dhcpv4_options      : []
dhcpv6_options      : []

```

Reply:
Ahh, thats right.Thanks. Sticking to leader pod will alleviate the use of certs to be used via other pods. Make sense. Also #1245 would make sense in future use cases for sure where containers need to leveraged for execution.